amp.php

```diff
@@ -251,4 +251,4 @@ function _amp_incorrect_plugin_slug_admin_notice() {
 
 register_deactivation_hook( __FILE__, 'amp_deactivate' );
 
-amp_bootstrap_plugin();
+add_action( 'plugins_loaded', 'amp_bootstrap_plugin', 8 );
```

Comment:
Is this too late for the services to initialize properly?

Note that `ReaderThemeLoader` needs to run at `plugins_loaded` priority 9, so that is why 8 is here.

Reply:
The only thing that could break by that would be if a Service needs to load before another plugin is loaded, and optimistically depends on the alphabetical loading order of the plugins. That's not something we should do, so I'd say we're safe with that action.

Reply:
Oh, just realized this will fail for `PluginSuppression` as it stands at present:

https://github.com/ampproject/amp-wp/blob/3e613fc0cfd9bdcc770c77c43fabb99f569c90e0/src/PluginSuppression.php#L53-L63

Reply:
Also `AmpSlugCustomizationWatcher`:

https://github.com/ampproject/amp-wp/blob/3e613fc0cfd9bdcc770c77c43fabb99f569c90e0/src/AmpSlugCustomizationWatcher.php#L38-L40

Reply:
So, this could be changed to:

```suggestion
add_action( 'plugins_loaded', 'amp_bootstrap_plugin', defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : ~PHP_INT_MAX );
```

And then `PluginSuppression` would need to be changed to add `1` to the `$priority` (so it is `PHP_INT_MIN + 1`).

The requirement then for someone using `amp_is_enabled` would be for them to add it _before_ the `plugins_loaded` action is done.